### ISSUE TITLE
Cloud-to-Cloud difference - inconsistent color changes

### ISSUE THREAD

**Issue by mathib** [CloudCompare/CloudCompare#626]
## Current Behavior
When comparing a two point clouds (one from a laser scanner and one subsampled on an CAD object in .OBJ) the resulting 3D color map, contains inconsistent colors: when the "show NaN/out of range values in grey" (SF display params > Parameters) is switched OFF the visual results (3D color map) are not the same as the ones showed in the histogram. If the same parameter is switched ON, the visual results look OK.

## Expected Behavior
There should be in my case no difference when the above parameter is switched ON or OFF, because there are no NaN or out of range values.

## Context/Comments
I assumed it would have to be something with the shader and the graphical card, so I tried to switch of the "enable shader for faster display" under Display > Display Options > color scale (scalar field). When this function was turned off, the 3D color map looks correct both when the "show NaN/out of range values in grey" is turned ON or OFF. I don't know if this is normal behavior. I also think that the "enable shader" option was ON by default.

## Your Environment
* OS & Version: Windows 10 Enterprise
* CC Version: 2.8.1
* Graphics card: NVIDIA Quadro M2000M

**Comment by dgirardeau** [comment]
Interesting. Can you maybe send me the project? (a BIN file containing the cloud with the distances)? (cloudcompare [at] danielgm.net)

**Comment by jarett-lee** [comment]
## Current Behavior
[Imgur link](https://imgur.com/a/xj0ql)

## Context/Comments
Notably the color changes when converted to RGB, indicating the lighting or shading is incorrect in the unchecked - show NaN/out of range values in grey.

The cloud is about 97 MB, which is on the large side, but I can probably find a much smaller example if you want one.

## Your Environment
- OS & Version: Windows 10 Home
- CC Version: 2.9.1
- Graphics card: NVIDIA GeForce GTX 1060

**Comment by dgirardeau** [comment]
Indeed, I haven't posted the (partial) answer here, but after discussion with 'mathib' later by email, we found the culprit:

```What I realized (on my side) is that the issue seems to be an interaction between the scalar field display shader and the normals.```

If you disable the normals (or the shader), the problem shouldn't occur.

We just have to fix it now ;)
